src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -109,24 +107,13 @@ public async Task<ParamsSemanticModel> CompileParams(string inputPath, bool skip
             {
                 throw new Exception($"Unable to read file {inputPath}");
             }
-            
             var paramsFile = SourceFileFactory.CreateBicepParamFile(inputUri, fileText);
 
-            var usingDeclarations = paramsFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
-
-            if(!PathHelper.TryGetUsingPath(usingDeclarations.FirstOrDefault(), out var bicepFilePath, out var failureBuilder))
-            {                
-                var message = failureBuilder(new DiagnosticBuilder.DiagnosticBuilderInternal(new TextSpan(0, 0))).Message;
-                
-                throw new BicepException(message);
-            }
-
-            Uri.TryCreate(inputUri, bicepFilePath, out var bicepFileUri);
-
-            var bicepCompilation = await CompileAsync(bicepFileUri!, skipRestore);
+            var model = new ParamsSemanticModel(paramsFile, (Uri uri) => {
```

Comment:
Any thoughts on this? @jeskew

Reply:
It's normally a bad idea to run async code synchronously, but I don't think you have much of a choice here (unless the return value for any ParamsSemanticModel method that depends on the compiled template were made async). I believe `(Uri uri) => CompileAsync(uri, skipRestore).GetAwaiter().GetResult()` is the *preferred* way to run async code synchronously.